cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -1020,19 +1018,19 @@ std::shared_ptr<Array> MakeComplex128(const std::shared_ptr<Array>& real,
 }
 
 std::shared_ptr<Array> ExampleUuid() {
-  auto arr = ArrayFromJSON(
+  auto arr = arrow::ArrayFromJSON(
```

Comment:
```suggestion
  auto arr = json::ArrayFromJSONString(
```

Reply:
Is this change and other similar ones below correct? I think what I have is correct but maybe am misunderstanding.

Reply:
My thinking was this file provides `*ArrayFromJSON` alias for existing test code, so the inner implementation should be based on the public `*ArrayFromJSONString` methods. A number of methods above follow that rule.

Makes sense to have those methods that do not return a `Result` to not duplicate the `Result`-unwrapping.

What is your methodology here?

Reply:
My methodology was just to keep the patch here small and focused. I think I see your point about refactoring the internals though. What do you think about doing that as a follow-up issue to this one?

Reply:
I was just thinking a file should be consistent, either use the new or the old methods. This file provides some convenience methods for testing, so it doesn't fully fit that aim. I am happy to withdraw my comments if you think the state of this file is sound.